### ISSUE TITLE
Strange behaviour with custom backends compiled against different DeviceAccess version

### ISSUE THREAD

**Issue by mhier** [ChimeraTK/DeviceAccess-PythonBindings#1]
When using a custom backend (e.g. by LD_PRELOADing it), which has been compiled against a different DeviceAccess version than the PythonBindings, I have observed that all registers are read as 0, even if they are set to non-zero values in the backend constructor. When compiling the PythonBindings against the same DeviceAccess version as the backend, everything works as expected.

When trying the same thing with QtHardMon (which is normally compiled against the same DeviceAccess version as the PythonBindings on my system), I get a useful error message telling me that I am mixing DeviceAccess versions. I would expect the same error message from the PythonBindings. I don't understand why it does not appear, since the error message is coming from DeviceAccess, as far as I know. Maybe the PythonBindings are circumventing the foreseen backend creation method in the factory?

Also this issue raises the question, whether the PythonBindings (and also QtHardMon) shouldn't be released as new Debian packages whenever there is a new DeviceAccess version. Otherwise the use of custom backends in Python scripts is practically impossible most of the time (without compiling the PythonBindings manually).

**Comment by killenb** [comment]
ldpreload is deprecated. Do you get the error with @LOADLIB in the dmap
file? According re-releasing python bindings I agree, but as usual its more
complicated then that.
Am 10.07.2017 17:21 schrieb "Martin Hierholzer" <notifications@github.com>:

**Comment by mhier** [comment]
What is the exact syntax? I cannot find any documentation about this, also it seems not to be tested...

**Comment by mhier** [comment]
Right, using that new feature it works. Unfortunately this feature is currently completely undocumented, isn't it?
